src/Core/BackgroundSchedulePool.cpp

```diff
@@ -214,6 +215,11 @@ BackgroundSchedulePool::~BackgroundSchedulePool()
             std::lock_guard lock_tasks(tasks_mutex);
             std::lock_guard lock_delayed_tasks(delayed_tasks_mutex);
 
+            // Notify all delayed tasks that the owning BackgroundSchedulePool is being destroyed.
+            // This prevents any further use of the pool (e.g., cancelDelayedTask) from within
+            // BackgroundSchedulePoolTaskInfo::deactivate(), avoiding use-after-free errors.
+            for (auto & task : delayed_tasks)
+                task.second->pool_shutdown = true;
```

Comment:
Here we access `pool_shutdown` without locking the corresponding `schedule_mutex`. This is a data race.